### PR TITLE
Do not call destructor if Marine was not initialized properly

### LINKED ISSUE #18: Don't call Marine destructor when Marine wasn't initialized properly
If something in `__init__` fails and `self._marine` is not properly loaded, the destructor still calls `self._marine.destroy_marine()` and fails

### PR DESCRIPTION
**Pull request by domerd**
Closes issue #18

### CODE REVIEW

File: marine/marine.py [hunk 1]
```diff
@@ -153,4 +153,5 @@ def _add_or_get_filter(
         return filter_id, err_msg_value
 
     def __del__(self):
-        self._marine.destroy_marine()
+        if hasattr(self, '_marine') and self._marine:
```
Comment (yehonatanz) [inline marine/marine.py]: Can be simply:
```python
if getattr(self, "_marine"):
    ...
```

**Review by yehonatanz — CHANGES_REQUESTED** [review]
Have you ran black on the code?

File: marine/marine.py (hunk 1 above)
Comment (domerd) [reply marine/marine.py]: Fixed + Ran Black

Comment (yehonatanz) [reply marine/marine.py]: Oops I meant:
```python
if getattr(self, "_marine", None):
    ...
```

**Review by yehonatanz — APPROVED** [review]

**Review by tomlegkov — APPROVED** [review]

**Pull request merged**